### PR TITLE
frog steps updated

### PR DESCRIPTION
**Pull request by joynewton1996**

### CODE REVIEW

File: hacker_Rank/May5_Answer.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        if 1 <= a[i] <= x:
+            for j in range(len(a)):
+                if a[i] == a[j] and a[i] not in b:
+                    b.append(a[i])
+                    seconds = j
+
+    print("fianl", b)
+    return "seconds " + str(seconds) + "  " + "Number of steps:" + str(b)
+
+
```
Comment (roy-einstein) [inline hacker_Rank/May5_Answer.py:24]: a=[1,2,3,2,4,5,3]
x=6
earliest_time(a,x)

output :

[1, 2, 3, 2, 4, 5, 3]
fianl [1, 2, 3, 4, 5]
'seconds 5  Number of steps:[1, 2, 3, 4, 5]'

expected output : -1

**Review by roy-einstein — CHANGES_REQUESTED** [review]
Its not handling false case

File: hacker_Rank/May5_Answer.py (hunk 1 above)
Comment (joynewton1996) [reply hacker_Rank/May5_Answer.py:24]: updated

Comment (roy-einstein) [reply hacker_Rank/May5_Answer.py:24]: take input :
a=[2,2,3,2,4,5,3]
x=5
earliest_time(a,x)

output : 
[2, 2, 3, 2, 4, 5, 3]
'seconds 5  Number of steps:[2, 3, 4, 5]'

expected :
 -1

**Review by roy-einstein — CHANGES_REQUESTED** [review]
Code is still need to modify,

File: hacker_Rank/May5_Answer.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+def earliest_time(a, x):
+    print(a)
+    b = []
+    ref = []
+    for i in range(1, x + 1):
+        ref.append(i)
+    print(ref)
```
Comment (roy-einstein) [inline hacker_Rank/May5_Answer.py]: Use List comprehension

File: hacker_Rank/May5_Answer.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    for i in range(len(a)):
+        if a[i] == x or a[i] <= x:
+            for j in range(len(a)):
+                if a[i] == a[j] and a[i] not in b:
+                    b.append(a[i])
+                    seconds = j
+
+    if len(ref) == len(b):
```
Comment (roy-einstein) [inline hacker_Rank/May5_Answer.py]: this is incorrect way of checking, 
Use Sets, method, or element wise comparsion

**Review by roy-einstein — CHANGES_REQUESTED** [review]
needs improvement

File: hacker_Rank/__init__.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+from May4_Answer import array_rotation
+
```
Comment (roy-einstein) [inline hacker_Rank/__init__.py:2]: import all algorithms

**Review by roy-einstein — CHANGES_REQUESTED** [review]

File: hacker_Rank/May5_Answer.py [hunk 5]
```diff
@@ -2,6 +2,7 @@ def earliest_time(a, x):
     print(a)
     b = []
     ref = []
+    count = 0
     for i in range(1, x + 1):
         ref.append(i)
     print(ref)
```
Comment (roy-einstein) [inline hacker_Rank/May5_Answer.py]: use list compression

File: hacker_Rank/May5_Answer.py [hunk 6]
```diff
@@ -13,15 +14,19 @@ def earliest_time(a, x):
                     b.append(a[i])
                     seconds = j
 
-    if len(ref) == len(b):
+    for _ in ref:
+        if _ not in a:
+            count += 1
+
+    if count == 0:
```
Comment (roy-einstein) [inline hacker_Rank/May5_Answer.py]: wrong implantation 
using "_" as variables is not correct

**Review by roy-einstein — CHANGES_REQUESTED** [review]
rejecting the code

**Review by roy-einstein — APPROVED** [review]
approved

**Pull request merged**